timesketch/lib/analyzers/feature_plugins/winevt_features.py

```diff
@@ -0,0 +1,204 @@
+# Copyright 2023 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""This file contains the plugin for Windows event logs feature extraction."""
+
+import logging
+from typing import List, Dict
+
+from timesketch.lib.analyzers import interface as base_interface
+from timesketch.lib.analyzers.feature_plugins import interface
+from timesketch.lib.analyzers.feature_plugins import manager
+
+logger = logging.getLogger("timesketch.analyzers.feature")
+
+
+class WindowsEventFeatureExtractionPlugin(interface.BaseFeatureExtractionPlugin):
+    """A plugin for Windows event log feature extraction."""
+
+    NAME = "winevt_feature_extraction"
+    DISPLAY_NAME = "Windows Event Log Feature Extraction"
+    DESCRIPTION = (
+        "This plugin extracts Windows event logs attributes from plaso output"
+        " attribute `strings`"
+    )
+
+    EVENT_FIELDS = ["strings"]
+
+    def run_plugin(self, name: str, config: dict) -> str:
+        """Extracts features from events.
+
+        Args:
+            name (str): Feature extraction name.
+            config (dict): A dict that contains the configuration for the feature
+                extraction.
+
+        Returns:
+            str: Returns summary of the feature extraction.
+        """
+        return self.extract_features(name, config)
+
+    def validate_feature_config(self, name: str, config: dict) -> None:
+        """Validates the name and configuration.
+
```

Comment:
Moving the comment from line 95-99 here. A bit rephrased.

```suggestion
           If any of the required properties does not exist or is of the wrong
           type, a ValueError is raised and the feature definition is not extracted.
           This does not affect other feature extractions since each feature is
           executed in their own celery task.

```